File: client-norx/app/src/main/java/ua/in/meugen/myapplication/app/di/modules/AppModule.java
package ua.in.meugen.myapplication.app.di.modules;

import android.app.Application;
import android.content.Context;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import ua.in.meugen.myapplication.app.MyApp;
import ua.in.meugen.myapplication.app.di.qualifiers.AppContext;
import ua.in.meugen.myapplication.app.di.scopes.PerApplication;

@Module
public abstract class AppModule {

    @Binds  @PerApplication
    abstract Application bindApplication(final MyApp app);

    @Binds @AppContext @PerApplication
    abstract Context bindAppContext(final Application app);

    @Provides @PerApplication
    static ExecutorService provideExecutorService() {
        return Executors.newCachedThreadPool();
    }
}

File: client-coroutines/app/src/main/java/ua/meugen/android/client/coroutines/app/di/modules/AppModule.kt
package ua.meugen.android.client.coroutines.app.di.modules

import android.app.Application
import android.content.Context
import dagger.Binds
import dagger.Module
import dagger.android.support.AndroidSupportInjectionModule
import ua.meugen.android.client.coroutines.app.ClientCoroutines
import javax.inject.Named
import javax.inject.Singleton

const val APP_CONTEXT = "appContext"

@Module(includes = [AndroidSupportInjectionModule::class,
        NetworkModule::class, ActivitiesModule::class])
abstract class AppModule {

    @Binds @Singleton
    abstract fun bindApplication(clientCoroutines: ClientCoroutines): Application

    @Binds @Named(APP_CONTEXT) @Singleton
    abstract fun bindContext(application: Application): Context
}